frontend/src/components/AdminLink.vue

```diff
@@ -0,0 +1,29 @@
+<script setup lang="ts">
+import { apiGet } from "../api";
+import { computed } from "vue";
+import type { QueryFunction } from "react-query/types/core";
+import { useQuery } from "vue-query";
+import { useUserStore } from "../stores/user";
+
+const userStore = useUserStore();
+
+const url = `/api/v1/admin`;
+
+/*eslint-disable */
+const { isLoading, isError, data, error } = useQuery(
+  url,
+  apiGet as QueryFunction<boolean>,
+  {
+    retry: false,
+  }
+);
+/*eslint-enable */
+
+const isAdmin = computed(() => data );
```

Comment:
You could also do `const { isLoading, isError, data: isAdmin, error } = useQuery(` and this way you don't need to define `isAdmin` as a derivative of `data`.